src/test/java/org/openrewrite/kotlin/KotlinTypeMappingTest.java

```diff
@@ -510,6 +509,47 @@ public K.When visitWhen(K.When when, AtomicBoolean found) {
             );
         }
 
+        @Test
+        void operatorOverload() {
```

Comment:
This is a little odd -- the FIR associated with a KtBinaryExpression is the method invocation, so the K.BinaryType is a MethodType.